dna/modify.go

```diff
@@ -133,6 +139,23 @@ func Insert(seq []Base, insPos int64, insSeq []Base) []Base {
 	return append(seq[:insPos], append(insSeq, seq[insPos:]...)...)
 }
 
+// InsertStable is identical to Insert, but conserves the
+// underlying memory of the input destSeq slice if possible
+func InsertStable(destSeq []Base, insPos int, insSeq []Base) ([]Base, error) {
+	if insPos < 0 || insPos > len(destSeq) {
+		return nil, ErrInvalidPosition
+	}
+	if len(destSeq)+len(insSeq) > cap(destSeq) {
+		return Insert(destSeq, int64(insPos), insSeq), ErrExceedCapNewSliceCreated
+	}
+	destSeq = destSeq[:len(destSeq)+1]                                           // extend the LEN of the slice to include the inserted bases
+	copy(destSeq[insPos+len(insSeq):cap(destSeq)], destSeq[insPos:cap(destSeq)]) // shift bases in slice by len(insSeq) at the position of the insertion
+	for i := 0; i < len(insSeq); i++ {                                           // add in the inserted bases
+		destSeq[insPos+i] = insSeq[i]
```

Comment:
Nothing wrong with this for-loop, but this loop could be replaced by another copy command:
copy(destSeq[insPos:insPos+len(insSeq)], insSeq)